model/PermissionsService.php

```diff
@@ -55,26 +55,36 @@ public function __construct(
         $this->eventManager = $eventManager;
     }
 
-    public function savePermissions(
+    public function saveResourcePermissions(
         bool $isRecursive,
-        core_kernel_classes_Class $class,
+        core_kernel_classes_Resource $resource,
         array $privilegesToSet
-    ): void {
-        $currentPrivileges = $this->dataBaseAccess->getResourcePermissions($class->getUri());
-
+    ):void {
```

Comment:
Boolean arguments for public apis are not a good practice. Explicit method declaration is better: https://martinfowler.com/bliki/FlagArgument.html

If you still want to maintain it, I would at least move it to the end of the parameter list. Wdyt?

Reply:
Ok. This is my proposition: 
https://github.com/oat-sa/extension-tao-dac-simple/pull/176/commits/e53b1c70d35edf88c8435968c4be15c25e7762e8

Reply:
Here is another approach:
https://github.com/oat-sa/extension-tao-dac-simple/pull/176/commits/262b392984dce9a021b263040cda70ebf53270c3